File: skill-graph-tmp/fp.js
#!/usr/bin/env node
const fs = require('fs');
const path = require('path');
const crypto = require('crypto');

process.chdir('/home/runner/work/aeon/aeon');

const buf = [];

for (const fn of ['aeon.yml', 'skills.json']) {
  const data = fs.readFileSync(fn);
  const digest = crypto.createHash('sha1').update(data).digest('hex');
  buf.push(`${digest}  ${fn}`);
}

const skillDirs = fs.readdirSync('skills').filter(d => {
  return fs.statSync(path.join('skills', d)).isDirectory();
}).sort();

const skillFiles = skillDirs
  .map(d => `skills/${d}/SKILL.md`)
  .filter(p => fs.existsSync(p));

const memRe = /memory\/(?:topics|state)\/[a-zA-Z0-9_.-]+/g;

for (const f of skillFiles) {
  const text = fs.readFileSync(f, 'utf8');
  const lines = text.split('\n');
  let n = 0;
  for (const line of lines) {
    if (line === '---') { n++; continue; }
    if (n === 1) buf.push(`${f}: ${line}`);
  }
  // emulate grep -hE '^depends_on:|^- skill:|consume:|parallel:|trigger:'
  // grep BRE/ERE alternation: each alternative is matched independently — first 2 are anchored, rest unanchored
  for (const line of lines) {
    if (
      /^depends_on:/.test(line) ||
      /^- skill:/.test(line) ||
      /consume:/.test(line) ||
      /parallel:/.test(line) ||
      /trigger:/.test(line)
    ) {
      buf.push(line);
    }
  }
  // grep -hoE 'memory/(topics|state)/...' | sort -u
  const found = new Set();
  let m;
  memRe.lastIndex = 0;
  while ((m = memRe.exec(text)) !== null) found.add(m[0]);
  for (const r of [...found].sort()) buf.push(r);
}

const innerInput = buf.join('\n') + '\n';
const innerSha = crypto.createHash('sha1').update(innerInput).digest('hex');
console.log(`${innerSha}  -`);

File: skill-graph-tmp/fp.py
#!/usr/bin/env python3
"""Replicate the bash fingerprint logic from skills/skill-graph/SKILL.md step 1."""
import hashlib, os, re, glob

os.chdir('/home/runner/work/aeon/aeon')

buf = []
for fn in ['aeon.yml', 'skills.json']:
    with open(fn, 'rb') as f:
        digest = hashlib.sha1(f.read()).hexdigest()
    buf.append(f'{digest}  {fn}')

skill_files = sorted(glob.glob('skills/*/SKILL.md'))
mem_re = re.compile(r'memory/(?:topics|state)/[a-zA-Z0-9_.-]+')
grep_re = re.compile(r'(^depends_on:)|(^- skill:)|(consume:)|(parallel:)|(trigger:)', re.M)

for f in skill_files:
    with open(f, 'r', encoding='utf-8', errors='replace') as fp:
        text = fp.read()
    n = 0
    for line in text.split('\n'):
        if line == '---':
            n += 1
            continue
        if n == 1:
            buf.append(f'{f}: {line}')
    for m in grep_re.finditer(text):
        ls = text.rfind('\n', 0, m.start()) + 1
        le = text.find('\n', m.start())
        if le < 0:
            le = len(text)
        buf.append(text[ls:le])
    refs = sorted(set(m.group(0) for m in mem_re.finditer(text)))
    buf.extend(refs)

inner_input = ('\n'.join(buf) + '\n').encode()
inner_sha = hashlib.sha1(inner_input).hexdigest()
print(f'{inner_sha}  -')
